project-koku/koku
# COST-1161: Source async delete follow ups

This change stops ingest of pending delete sources and also allows for the user to re-add the source back before the provider is completely removed.

**Testing**
1. Create a source shutdown the worker then delete the source.  Verify that the provider is not active
2. POST /source-status for the source should return a 404 since we do not want to re-activate a provider if it's pending deletion.
3. Add another source with duplicate source name, billing source, and authenticaiton.  Verify creation is successful.  Even though the first source is still pending delete.
4. Start worker and verify that first source is remvoed. And second completes ingest.

**Test Results**
[cost_1161_ut.txt](https://github.com/project-koku/koku/files/6183169/cost_1161_ut.txt)
